pype/lib.py

```diff
@@ -708,3 +709,607 @@ def __init__(self):
+    }, {
+        "asset_entity": <AssetEntity2>,
+        "containers": [<Container3>, ...]
+    }, {
+        ...
+    }]
+    """
+    io.install()
+
+    # Get current asset name and entity
+    current_asset_name = io.Session["AVALON_ASSET"]
+    current_asset_entity = io.find_one({
+        "type": "asset",
+        "name": current_asset_name
+    })
+
+    # Skip if asset was not found
+    if not current_asset_entity:
+        print("Asset entity with name `{}` was not found".format(
+            current_asset_name
+        ))
+        return
+
+    # Prepare available loaders
+    loaders_by_name = {}
+    for loader in avalon.api.discover(avalon.api.Loader):
+        loader_name = loader.__name__
+        if loader_name in loaders_by_name:
+            raise KeyError("Duplicated loader name {0}!".format(loader_name))
+        loaders_by_name[loader_name] = loader
+
+    # Skip if there are any loaders
+    if not loaders_by_name:
+        print("There are no registered loaders.")
+        return
+
+    # Get current task name
+    current_task_name = os.environ["AVALON_TASK"]
+    current_task_name_low = current_task_name.lower()
+    # Load workfile presets for task
+    workfile_presets = get_workfile_build_presets(current_task_name_low)
+
+    # Skip if there are any presets for task
+    if not workfile_presets:
+        log.warning(
+            "Current task `{}` does not have any loading preset.".format(
+                current_task_name
+            )
+        )
+        return
+
+    # Get presets for loading current asset
+    current_context = workfile_presets.get("current_context")
+    # Get presets for loading linked assets
+    link_context = workfile_presets.get("linked_assets")
+    # Skip if both are missing
+    if not current_context and not link_context:
+        log.warning("Current task `{}` has empty loading preset.".format(
+            current_task_name
+        ))
+        return
+
+    elif not current_context:
+        log.warning((
+            "Current task `{}` doesn't have any loading preset for it's context."
+        ).format(current_task_name))
+
+    elif not link_context:
+        log.warning((
+            "Current task `{}` doesn't have any"
+            "loading preset for it's linked assets."
+        ).format(current_task_name))
+
+    # Prepare assets to process by workfile presets
+    assets = []
+    current_asset_id = None
+    if current_context:
+        # Add current asset entity if preset has current context set
+        assets.append(current_asset_entity)
+        current_asset_id = current_asset_entity["_id"]
+
+    if link_context:
+        # Find and append linked assets if preset has set linked mapping
+        link_assets = get_linked_assets(current_asset_entity)
+        if link_assets:
+            assets.extend(link_assets)
+
+    # Skip if there are any assets
+    # - this may happend if only linked mapping is set and there are not links
```

Comment:
Some typos in comment

```
# Skip if there are no assets. This can happen if only linked mapping is set
# and there are no links for his asset.
```